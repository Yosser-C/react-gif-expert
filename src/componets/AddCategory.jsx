import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setinputValue] = useState("");

  const onInputChange = ({ target }) => {
    setinputValue(target.value);
  };

  const onsubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    //setCategories((categories) => [inputValue, ...categories]);
    setinputValue("");
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onsubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.PropTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
