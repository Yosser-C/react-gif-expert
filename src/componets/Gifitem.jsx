import propTypes from "prop-types";

export const Gifitem = ({ title, url, id }) => {
  console.log({ url, title, id });

  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

Gifitem.prototype = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
};
