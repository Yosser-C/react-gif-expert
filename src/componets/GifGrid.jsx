import { Gifitem } from "./Gifitem";
import { useFetchGifs } from "../Hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);
  //console.log({ isLoading });
  // dasjkdhj
  //.then(newImages => setImages(newImages))

  return (
    <>
      <h3>{category}</h3>

      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {/* images.map...      title={image.title} url={image.url}*/}
        {images.map((image) => (
          <Gifitem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.PropTypes = {
  category: PropTypes.string.isRequired,
};
