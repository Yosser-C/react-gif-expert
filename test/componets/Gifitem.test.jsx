import { render, screen } from "@testing-library/react";
import { Gifitem } from "../../src/componets/Gifitem";

describe("pruebas en Gifitem", () => {
  const title = "saitama";
  const url = "https://one puch.com/onetock.jpg";
  test("debe de ahcer match con el snapshot", () => {
    const { container } = render(<Gifitem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("debe mostrar la imagen con el url y el alt indicado ", () => {
    render(<Gifitem title={title} url={url} />);
    //screen.debug();
    //expect(screen.getByRole("img").src).toBe(url);
    //expect(screen.getByRole("img").alt).toBe(title );
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });
  test("debe de mostrar el titulo en el componente", () => {
    render(<Gifitem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy(); //que exista el titulo
  });
});
