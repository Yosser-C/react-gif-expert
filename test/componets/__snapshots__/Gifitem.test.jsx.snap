// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`pruebas en Gifitem debe de ahcer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one puch.com/onetock.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
